lib/oban/config.ex

```diff
@@ -91,168 +133,188 @@ defmodule Oban.Config do
-      raise ArgumentError,
-            "expected :engine to be an Oban.Queue.Engine, got: #{inspect(engine)}"
+  defp validate_opt({:peer, peer}) do
+    if peer == false or Code.ensure_loaded?(peer) do
+      :ok
+    else
+      {:error, "expected :peer to be false or an Oban.Peer, got: #{inspect(peer)}"}
     end
   end
 
-  defp validate_opt!({:notifier, notifier}) do
-    unless Code.ensure_loaded?(notifier) and function_exported?(notifier, :listen, 2) do
-      raise ArgumentError,
-            "expected :notifier to be an Oban.Notifier, got: #{inspect(notifier)}"
+  defp validate_opt({:plugins, plugins}) do
+    if is_list(plugins) do
+      Validation.validate(plugins, &validate_plugin/1)
+    else
+      {:error, "expected :plugins to be a list, got: #{inspect(plugins)}"}
     end
   end
 
-  defp validate_opt!({:name, _}), do: :ok
-
-  defp validate_opt!({:node, node}) do
-    unless is_binary(node) and String.trim(node) != "" do
-      raise ArgumentError,
-            "expected :node to be a non-empty binary, got: #{inspect(node)}"
+  defp validate_opt({:prefix, prefix}) do
+    if is_binary(prefix) and Regex.match?(~r/^[a-z0-9_]+$/i, prefix) do
+      :ok
+    else
+      {:error, "expected :prefix to be an alphanumeric string, got: #{inspect(prefix)}"}
     end
   end
 
-  defp validate_opt!({:peer, peer}) do
-    unless peer == false or Code.ensure_loaded?(peer) do
-      raise ArgumentError,
-            "expected :peer to be false or Oban.Peer, got: #{inspect(peer)}"
+  defp validate_opt({:queues, queues}) do
+    if Keyword.keyword?(queues) do
+      Validation.validate(queues, &validate_queue/1)
+    else
+      {:error, "expected :queues to be a keyword list, got: #{inspect(queues)}"}
     end
   end
 
-  defp validate_opt!({:plugins, plugins}) do
-    unless is_list(plugins) do
-      raise ArgumentError, "expected :plugins to be a list, got #{inspect(plugins)}"
+  defp validate_opt({:repo, repo}) do
+    if Code.ensure_loaded?(repo) and function_exported?(repo, :config, 0) do
+      :ok
+    else
+      {:error, "expected :repo to be an Ecto.Repo, got: #{inspect(repo)}"}
     end
-
-    Enum.each(plugins, &validate_plugin/1)
   end
 
-  defp validate_opt!({:prefix, prefix}) do
-    unless is_binary(prefix) and Regex.match?(~r/^[a-z0-9_]+$/i, prefix) do
-      raise ArgumentError,
-            "expected :prefix to be a binary with alphanumeric characters, got: #{inspect(prefix)}"
-    end
+  defp validate_opt({:shutdown_grace_period, period}) do
+    Validation.validate_integer(:shutdown_grace_period, period)
   end
 
-  defp validate_opt!({:queues, queues}) do
-    unless Keyword.keyword?(queues) and Enum.all?(queues, &valid_queue?/1) do
-      raise ArgumentError,
-            "expected :queues to be a keyword list of {atom, integer} pairs or " <>
-              "a list of {atom, keyword} pairs, got: #{inspect(queues)}"
+  @log_levels ~w(false emergency alert critical error warning warn notice info debug)a
```

Comment:
What do you about putting this at the top, right before the `@type` for example?

Reply:
In favor. Done 👍